host/sgx/linux/exception.c

```diff
@@ -153,7 +153,10 @@ static void _register_signal_handlers(void)
     // Use sa_sigaction instead of sa_handler, allow catching the same signal as
     // the one you're currently handling, and automatically restart the system
     // call that interrupted the signal.
-    sig_action.sa_flags = SA_SIGINFO | SA_NODEFER | SA_RESTART;
+    // SA_ONSTACK is required in cases where this signal handler may be
+    // registered in a non-C language like Go with alternate stack
+    // implementations.
+    sig_action.sa_flags = SA_SIGINFO | SA_NODEFER | SA_RESTART | SA_ONSTACK;
```

Comment:
Is it effective without setting altstack via `sigaltstack`?

Reply:
For code called from Golang it is effective since the Go runtime calls `sigaltstack` when initializing an OS thread structure (an "M").  In fact, Golang requires that called non-Go code use the `SA_ONSTACK` flag with `sigaction` when installing signal handlers [[link]](https://pkg.go.dev/os/signal#hdr-Go_programs_that_use_cgo_or_SWIG).

Reply:
I think you're right, I need a `signaltstack`.  I've added one in to this PR.  Working on figuring out how to build from source to test this out, but thought I'd send the code for a look in parallel.

Reply:
Rolfe was correct as well, Golang already sets an alternate stack.  I've gone a new route and set ONSTACK by checking for an existing alternate stack and only setting it if one is found.

Reply:
Figured out this isn't necessary; within our binary we can use the pattern:

```
  struct sigaction act;
  sigaction(SIGPIPE, 0, &act);
  act.sa_flags |= SA_ONSTACK;
  sigaction(SIGPIPE, &act, 0);
```